mne/tests/test_chpi.py

```diff
@@ -210,53 +241,96 @@ def test_calculate_chpi_positions():
         with catch_logging() as log_file:
             _calculate_chpi_positions(raw_bad, t_step_min=1., verbose=True)
     # ignore HPI info header and [done] footer
-    assert '0/5 good' in log_file.getvalue().strip().split('\n')[-2]
+    assert '0/5 good HPI fits' in log_file.getvalue()
 
     # half the rate cuts off cHPI coils
     raw.info['lowpass'] /= 2.
     with pytest.raises(RuntimeError, match='above the'):
         _calculate_chpi_positions(raw)
 
-    # test on 5k artemis data
+
+def test_calculate_chpi_positions_artemis():
+    """Test on 5k artemis data."""
     raw = read_raw_artemis123(art_fname, preload=True)
     mf_quats = read_head_pos(art_mc_fname)
-    with catch_logging() as log:
-        py_quats = _calculate_chpi_positions(raw, t_step_min=2.,
-                                             verbose='debug')
-    _assert_quats(py_quats, mf_quats, dist_tol=0.004, angle_tol=2.5)
+    py_quats = _calculate_chpi_positions(raw, t_step_min=2., verbose='debug')
+    _assert_quats(py_quats, mf_quats, dist_tol=0.001, angle_tol=1.)
+
+
+def test_initial_fit_redo():
+    """Test that initial fits can be redone based on moments."""
+    raw = read_raw_fif(chpi_fif_fname, allow_maxshield='yes')
+    slopes = np.array(
+        [[c['slopes'] for c in raw.info['hpi_meas'][0]['hpi_coils']]])
+    amps = np.linalg.norm(slopes, axis=-1)
+    amps /= slopes.shape[-1]
+    assert_array_less(amps, 5e-11)
+    assert_array_less(1e-12, amps)
+    proj, _, _ = _setup_ext_proj(raw.info, ext_order=1)
+    chpi_amplitudes = dict(times=np.zeros(1), slopes=slopes, proj=proj)
+    chpi_locs = compute_chpi_locs(raw.info, chpi_amplitudes)
+
+    # check GOF
+    coil_gof = raw.info['hpi_results'][0]['goodness']
+    assert_allclose(chpi_locs['gofs'][0], coil_gof, atol=0.3)  # XXX not good
+
+    # check moment
+    # XXX our forward and theirs differ by an extra mult by _MAG_FACTOR
```

Comment:
@bloyl coil moments added. Feel free to look and merge if you're happy

As a side note, I don't know why our calculation and MaxFilter's differ by `_MAG_FACTOR` (1e-7). I'm guessing that they accidentally double-multiply by it? cc @agramfort in case you know.